### DISCUSSION TITLE
Does certbot run all renewal hooks for unchanged certificates on purpose?

### DISCUSSION THREAD

**Discussion by harridu** [certbot/certbot#9135] [Q&A] [answered]
Hi folks,

even if just a single certificate is renewed, certbot seems to run **all** post scripts defined for any certificate. This is pretty painful. Jenkins and Nexus are restarted, secrets are changed in some Kubernetes clusters, etc. Shouldn't certbot run only the post script for the changed certificate?

**Comment by zoracon** [comment]
Moving to Issues for better visibility

**Comment by ohemorange** [comment, marked as answer]
Answered in https://github.com/certbot/certbot/issues/9189
